README.md

```diff
@@ -2,5 +2,435 @@
+
+```kotlin
+dependencies {
+    implementation("team.razz.eva:eva-uow:$eva_version")
+    implementation("team.razz.eva:eva-repository:$eva_version")
+    implementation("team.razz.eva:eva-persistence-jdbc:$eva_version")
+}
+```
+
+### Model
+Define events for your business domain.
+You can read more about domain and integration events [here](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/domain-events-design-implementation). 
+```kotlin
+sealed class WalletEvent : ModelEvent<Wallet.Id> {
+
+    override val modelName = "Wallet"
+
+    data class Created(
+        override val modelId: Wallet.Id,
+        val currency: Currency,
+        val amount: ULong,
+        val expireAt: Instant
+    ) : WalletEvent(), ModelCreatedEvent<Wallet.Id> {
+        override fun integrationEvent() = buildJsonObject {
+            put("currency", currency.currencyCode)
+            put("amount", amount.toLong())
+            put("expireAt", expireAt.epochSecond)
+        }
+    }
+
+    data class Deposit(
+        override val modelId: Wallet.Id,
+        val walletAmount: ULong,
+        val depositAmount: ULong
+    ) : WalletEvent(), ModelCreatedEvent<Wallet.Id> {
+        override fun integrationEvent() = buildJsonObject {
+            put("walletAmount", walletAmount.toLong())
+            put("depositAmount", depositAmount.toLong())
+        }
+    }
+}
+```
+
+Define a model and methods that change model's state.
+On any model's modification we should raise an event about it.
+```kotlin
+class Wallet(
+    id: Id,
+    val currency: Currency,
+    val amount: ULong,
+    val expireAt: Instant,
+    entityState: EntityState<Id, WalletEvent>
+) : Model<Wallet.Id, WalletEvent>(id, entityState) {
+
+    data class Id(override val id: UUID) : ModelId<UUID>
+
+    fun deposit(toDeposit: ULong) = Wallet(
+        amount = amount - toDeposit,
+        currency = currency,
+        id = id(),
+        expireAt = expireAt,
+        entityState = entityState()
+            .raiseEvent(WalletEvent.Deposit(id(), amount, toDeposit))
+    )
+}
+```
+
+### Unit of work
+We need *queries* interface, so we can query our existing models
+```kotlin
+interface WalletQueries {
+    suspend fun find(id: Wallet.Id): Wallet?
+}
+```
+Now we can write our first unit of work.
+In our framework unit of work stands for Command in CQRS pattern.
+You can read more about CQRS [here](https://docs.microsoft.com/en-us/azure/architecture/patterns/cqrs) and [here](https://martinfowler.com/bliki/CQRS.html).
+Unit of work is a transactional operation.
+Here the unit of work either returns an existing wallet by ID or creates a new one and returns it.
+
+```kotlin
+class CreateWalletUow(
+    private val queries: WalletQueries,
+    clock: Clock
+) : UnitOfWork<ServicePrincipal, Params, Wallet>(clock) {
+
+    @Serializable
+    data class Params(val id: String, val currency: String) : UowParams<Params> {
```

Comment:
Why don't send typed id and currency in params?

Reply:
because UUID is not serializable by default, don't want in scope of this small example create serializer, use Contextual and etc

Reply:
BTW, need to explain why we ask params to be serialized and how it related to uow and model events.